slack-lint/src/main/java/slack/lint/denylistedapis/DenyListedApiDetector.kt

```diff
@@ -0,0 +1,459 @@
+          "the reactive domain."
+      ),
+      DenyListedEntry(
+        className = "kotlinx.coroutines.test.TestBuildersKt",
+        functionName = "runBlockingTest",
+        errorMessage =
+        "Use runBlocking or runBlockingWithTestDispatcher instead. runBlockingTest " +
+          "is often overkill. The only difference between it and runBlocking is that it gives you " +
+          "a TestDispatcher. Treat it like RxJava's TestScheduler: if you don't need it, leave it " +
+          "out. But even if you do need it, you are better off using TestDispatcher directly: if a " +
+          "coroutine is blocked on something other than TestDispatcher (which isn't uncommon - " +
+          "most things that block in Cash App aren't coroutines right now), at the end of your " +
+          "runBlockingTest block it will fail the test. This is not what you want."
+      ),
+      DenyListedEntry(
+        className = "androidx.viewpager2.widget.ViewPager2",
+        functionName = "setId",
+        parameters = listOf("int"),
+        arguments = listOf("ViewCompat.generateViewId()"),
+        errorMessage =
+        "Use an id defined in resources or a statically created instead of generating with ViewCompat.generateViewId(). See https://issuetracker.google.com/issues/185820237"
+      ),
+      DenyListedEntry(
+        className = "androidx.viewpager2.widget.ViewPager2",
+        functionName = "setId",
+        parameters = listOf("int"),
+        arguments = listOf("View.generateViewId()"),
+        errorMessage =
+        "Use an id defined in resources or a statically created instead of generating with View.generateViewId(). See https://issuetracker.google.com/issues/185820237"
+      ),
+      DenyListedEntry(
+        className = "java.util.LinkedList",
+        functionName = "<init>",
+        errorMessage =
+        "For a stack/queue/double-ended queue use ArrayDeque, for a list use ArrayList. Both are more efficient internally."
+      ),
+      DenyListedEntry(
+        className = "java.util.Stack",
+        functionName = "<init>",
+        errorMessage = "For a stack use ArrayDeque which is more efficient internally."
+      ),
+      DenyListedEntry(
+        className = "java.util.Vector",
+        functionName = "<init>",
+        errorMessage =
+        "For a vector use ArrayList or ArrayDeque which are more efficient internally."
+      ),
+      DenyListedEntry(
+        className = "io.reactivex.rxjava3.schedulers.Schedulers",
+        functionName = "newThread",
+        errorMessage =
+        "Use a scheduler which wraps a cached set of threads. There should be no reason to be arbitrarily creating threads on Android."
+      ),
+      DenyListedEntry(
+        className = "android.os.Build.VERSION_CODES",
+        fieldName = MatchAll,
+        errorMessage =
+        "No one remembers what these constants map to. Use the API level integer value directly since it's self-defining."
```

Comment:
Won't this conflict with the MagicNumber rule?

Reply:
hmmm, good point. I do like this though too. I'll disable this for now and revisit after looking more at magicnumber

Reply:
Disabled in 4c6b3bc